### ISSUE TITLE
Stop swallowing exceptions with 'There was an error while trying to load the gem"

### ISSUE THREAD

**Issue by grosser** [bundler/bundler#4279]
this is not very helpful when trying to debug why something failed

```
rake/rake_test_loader.rb" "test/**/*_test.rb"
/opt/boxen/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.11.2/lib/bundler/runtime.rb:80:in `rescue in block (2 levels) in require': There was an error while trying to load the gem 'query_diet'. (Bundler::GemRequireError)
	from /opt/boxen/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.11.2/lib/bundler/runtime.rb:76:in `block (2 levels) in require'
```

**Comment by segiddins** [comment]
@grosser have a reproducible example of this happening?

**Comment by indirect** [comment]
I believe we _just_ fixed this, and it will go out in 1.12.

**Comment by grosser** [comment]
example is putting `raise 'foo'` into a gemspec ... I'll recheck on 1.12

**Comment by grosser** [comment]
yeah, code is good now `\nThe error message was: #{e.message}.`

**Comment by grosser** [comment]
ah no ... that was a different place
fix is in https://github.com/bundler/bundler/blob/f9a3a5148e7902ae4e30e837060806067e9025f5/lib/bundler/errors.rb#L70-L73

**Comment by RochesterinNYC** [comment]
For reference, this is the PR that addressed it: #4187
